### PR TITLE
Add ORB full float32 support

### PR DESCRIPTION
**Pull request by rfezzani**
## Description

This PR completes #4684.
#4696 is a prerequisite for this PR.

## Checklist

- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

File: skimage/transform/pyramids.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 def _smooth(image, sigma, mode, cval, multichannel=None):
     """Return image with each channel smoothed by the Gaussian filter."""
-    smoothed = np.empty(image.shape, dtype=np.double)
+    smoothed = np.empty_like(image)
```
Comment (rfezzani) [inline skimage/transform/pyramids.py:10]: This is properly managed in #4696.

**Comment by hmaarrfk** [comment]
Maybe we should just pin cython 0.17.2?

**Comment by emmanuelle** [comment]
Hum good question about cython. @rfezzani did you identify that the problems with float32 are due to a new version of cython?

**Comment by emmanuelle** [comment]
@meeseeksdev backport to v0.17.x

**Comment by emmanuelle** [comment]
When we'll have type annotations we'll be able to test all our functions accepting floats to see if they behave well with float32...

**Comment by soupault** [comment]
I wonder if this PR solves https://github.com/scikit-image/scikit-image/issues/2529 . @rfezzani could you comment here, please?

**Comment by rfezzani** [comment]
@soupault, unfortunately, the failing test in #2529 still fails in single precision :confused: This test is mainly a non-regression test and it is designed for double precision input :(

**Pull request merged**